scripts/branch_track_creation.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""This is a script for creating branches and tracks"""
+
+import json
+import logging
+import os
+import sys
+
+import requests
+import yaml
+from git import Repo
+
+GITHUB_API_URL = "https://api.github.com"
+DEFAULT_REPO_OWNER = "canonical"
+MAIN_BRANCH_NAMES = ["main", "master"]
+GITHUB_TOKEN_NAME = "KUBEFLOW_BOT_TOKEN"
+
+HEADERS = {"content-type": "application/vnd.github.v3+json"}
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+def get_git_diff():
+    """() -> [str]
```

Comment:
I don't think you have to go super deep into this rabbit hole for this PR but instead of having it in the docstring you might as well use [python typing](https://docs.python.org/3/library/typing.html).